editor.planx.uk/src/pages/Preview/Questions.tsx

```diff
@@ -69,59 +69,62 @@ const Questions = ({ previewEnvironment }: QuestionsProps) => {
   const isStandalone = previewEnvironment === "standalone";
   const { createAnalytics, node } = useAnalyticsTracking();
   const [gotFlow, setGotFlow] = useState(false);
-  const isSingleSession =
+  const isUsingLocalStorage =
     useStore((state) => state.path) === ApplicationPath.SingleSession;
 
-  if (isSingleSession) {
-    // Use local storage for simple, non-Save&Return flows
-    useEffect(() => {
-      setPreviewEnvironment(previewEnvironment);
-      if (isStandalone) {
-        const state = getLocalFlow(id);
-        if (state) {
-          resumeSession(state);
-        }
+  // Initial setup
+  useEffect(() => {
+    setPreviewEnvironment(previewEnvironment);
+    if (!isStandalone) return;
+
+    if (isUsingLocalStorage) {
+      const state = getLocalFlow(id);
+      if (state) resumeSession(state);
+      createAnalytics(state ? "resume" : "init");
+      setGotFlow(true);
+    } else {
+      NEW.getLocalFlow(sessionId).then((state) => {
+        // session data is resumed by ./ResumePage.tsx
         createAnalytics(state ? "resume" : "init");
         setGotFlow(true);
-      }
-    }, []);
-
-    useEffect(() => {
-      if (gotFlow && isStandalone && id) {
-        setLocalFlow(id, {
-          breadcrumbs,
-          id,
-          passport,
-          sessionId,
-          govUkPayment,
-        });
-      }
-    }, [gotFlow, breadcrumbs, passport, sessionId, id, govUkPayment]);
-  } else {
-    // Use lowcalStorage for Save & Return flows
-    useEffect(() => {
-      setPreviewEnvironment(previewEnvironment);
-      if (isStandalone) {
-        NEW.getLocalFlow(sessionId).then((state) => {
-          // session data is resumed by ./ResumePage.tsx
-          createAnalytics(state ? "resume" : "init");
-          setGotFlow(true);
-        });
-      }
-    }, []);
-
-    useEffect(() => {
-      if (gotFlow && isStandalone && sessionId) {
-        NEW.setLocalFlow(sessionId, {
-          breadcrumbs,
-          id,
-          passport,
-          sessionId,
-          govUkPayment,
-        });
-      }
-    }, [gotFlow, breadcrumbs, passport, sessionId, id, govUkPayment]);
-  }
+      });
+    }
+  }, [
+    createAnalytics,
+    id,
+    isStandalone,
+    isUsingLocalStorage,
+    previewEnvironment,
+    resumeSession,
+    sessionId,
+    setPreviewEnvironment,
```

Comment:
These are exhaustive dependencies to satisfy `react-hooks/exhaustive-deps`. None of these values should change, so it's effectively the same as the previous `[]`.